### PR TITLE
[MAEB] Add safety checks for encodec model

### PR DESCRIPTION
**Pull request by AdnanElAssadi56**
If you add a model or a dataset, please add the corresponding checklist:

* [dataset checklist](https://embeddings-benchmark.github.io/mteb/contributing/adding_a_dataset/#submit-a-pr)
* [model checklist](https://embeddings-benchmark.github.io/mteb/contributing/adding_a_model/#submitting-your-model-as-a-pr)

### CODE REVIEW

File: mteb/models/model_implementations/encodec_model.py [hunk 1]
```diff
@@ -46,8 +46,10 @@ def get_audio_embeddings(
         **kwargs: Any,
     ) -> Array:
         import torchaudio
+        import numpy as np
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/encodec_model.py]: top-level

File: mteb/models/model_implementations/encodec_model.py [hunk 2]
```diff
@@ -76,22 +78,46 @@ def get_audio_embeddings(
                     array = resampler(array)
 
                 array = array.squeeze()
+                
+                # Handle edge case where squeeze results in 0-dim tensor
+                if array.dim() == 0:
+                    array = array.unsqueeze(0)
+                
+                # Ensure we have at least some samples
+                if len(array) == 0:
```
Comment (Samoed) [inline mteb/models/model_implementations/encodec_model.py]: I think it's better to filter out such audios in tasks

File: mteb/models/model_implementations/encodec_model.py [hunk 3]
```diff
@@ -76,22 +78,46 @@ def get_audio_embeddings(
...
+                # Need enough samples to produce at least 1 frame in the latent space
+                # Being more conservative with 1 second minimum to ensure encoder works
+                min_samples = self.sampling_rate  # 1 second minimum
+                if len(array) < min_samples:
+                    padding = torch.zeros(min_samples - len(array))
+                    array = torch.cat([array, padding])
+                
+                # Pad or truncate to fixed length for consistent batching
+                if len(array) > max_samples:
+                    array = array[:max_samples]
```
Comment (Samoed) [inline mteb/models/model_implementations/encodec_model.py:94]: You can process max size and padding with processor https://github.com/huggingface/transformers/blob/a7f29523361b2cc12e51c1f5133d95f122f6f45c/src/transformers/models/encodec/feature_extraction_encodec.py#L87-L91

File: mteb/models/model_implementations/encodec_model.py [hunk 4]
```diff
@@ -76,22 +78,46 @@ def get_audio_embeddings(
...
+                
+                # Encodec has multiple downsampling layers with total stride ~320
+                # Need enough samples to produce at least 1 frame in the latent space
+                # Being more conservative with 1 second minimum to ensure encoder works
+                min_samples = self.sampling_rate  # 1 second minimum
+                if len(array) < min_samples:
+                    padding = torch.zeros(min_samples - len(array))
+                    array = torch.cat([array, padding])
+                
+                # Pad or truncate to fixed length for consistent batching
```
Comment (Samoed) [inline mteb/models/model_implementations/encodec_model.py]: I think you'll truncate total number of samples in batch rather than length

File: mteb/models/model_implementations/encodec_model.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/encodec_model.py]: I would be fine with handling it, but we should probably add a warning log + warning

File: mteb/models/model_implementations/encodec_model.py (hunk 3 above)
Comment (AdnanElAssadi) [reply mteb/models/model_implementations/encodec_model.py:94]: I don't think processor allows setting both truncation and padding to True.

Comment (Samoed) [reply mteb/models/model_implementations/encodec_model.py:94]: It will allow. 

```
padding (`bool`, `str` or [`~utils.PaddingStrategy`], *optional*, defaults to `True`):
                Select a strategy to pad the returned sequences (according to the model's padding side and padding
                index) among:

                - `True` or `'longest'`: Pad to the longest sequence in the batch (or no padding if only a single
                  sequence if provided).
                - `'max_length'`: Pad to a maximum length specified with the argument `max_length` or to the maximum
                  acceptable input length for the model if that argument is not provided.
            truncation (`bool`, *optional*, defaults to `False`):
                Activates truncation to cut input sequences longer than `max_length` to `max_length`.
```

Comment (AdnanElAssadi) [reply mteb/models/model_implementations/encodec_model.py:94]: And padding to max_length wouldn't exactly work because we need to also ensure a min_length when audio is too short. If the audio samples were too short, and after Encodec's multiple downsampling layers (stride), there will not be enough samples to produce even a single latent frame, resulting in a tensor with 0 elements in the time dimension.

And, by the way, it seems that the dataloader is passing empty batches when empty audio samples are not filtered. This can be triggered with the VoiceGenderClustering Dataset.

Comment (AdnanElAssadi) [reply mteb/models/model_implementations/encodec_model.py:94]: Lines 138:
`
        if padding and truncation:
            raise ValueError("Both padding and truncation were set. Make sure you only set one.")
`

Comment (Samoed) [reply mteb/models/model_implementations/encodec_model.py:94]: I see. I think you can use `padding` `True` by default and it would be enough

Comment (AdnanElAssadi) [reply mteb/models/model_implementations/encodec_model.py:94]: But we need both padding and truncation and a min_length. I think it is okay now.

**Comment by Samoed** [comment]
Can you fix linting issue? Seems that you're not using numpy now

**Comment by AdnanElAssadi** [comment]
Ran lint
